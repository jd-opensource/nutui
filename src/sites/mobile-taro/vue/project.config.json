{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
