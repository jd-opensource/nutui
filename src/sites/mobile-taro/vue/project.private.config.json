{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}