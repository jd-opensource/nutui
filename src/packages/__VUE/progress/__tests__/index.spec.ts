import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import Progress from '../index.vue';

test('should render progress when use props', async () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 100
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render different height and color when use color height props', async () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      strokeColor: 'blue',
      strokeWidth: '20',
      textColor: 'red'
    }
  });
  await nextTick();
  const inner = wrapper.find<HTMLElement>('.nut-progress-inner');
  expect(inner.element.style.background).toEqual('blue');
  const outer = wrapper.find<HTMLElement>('.nut-progress-outer');
  const span = wrapper.find<HTMLElement>('.nut-progress-text span');
  expect(span.element.style.color).toEqual('red');
  expect(outer.element.style.height).toEqual('20px');
});

test('should hide percentage when use showText props', () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      showText: false
    }
  });
  const text = wrapper.find<HTMLElement>('.nut-progress-text');
  expect(text.exists()).toBe(false);
});

test('should render inside percentage when use textInside props', () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      textInside: true
    }
  });
  const text = wrapper.find<HTMLElement>('.nut-progress-insidetext');
  expect(text.exists()).toBe(true);
});

test('should render custom size when use size props', () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      size: 'large'
    }
  });
  const text = wrapper.find<HTMLElement>('.nut-progress-large');
  expect(text.exists()).toBe(true);
});
