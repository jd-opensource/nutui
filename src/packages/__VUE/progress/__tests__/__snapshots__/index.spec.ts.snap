// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render progress when use props 1`] = `
"<div class=\\"nut-progress\\">
  <div class=\\"nut-progress-outer nut-progress-outer-part nut-progress-base\\">
    <div class=\\"nut-progress-inner\\" style=\\"width: 100%;\\"></div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class=\\"nut-progress-text\\"><span>100%</span></div>
</div>"
`;
